libs/barista-components/select/src/select.ts

```diff
@@ -524,10 +524,18 @@ export class DtSelect<T> extends _DtSelectMixinBase
       });
 
     this.options.changes
-      .pipe(startWith(null), takeUntil(this._destroy))
-      .subscribe(() => {
+      .pipe(
+        startWith(null),
+        switchMap(() =>
+          merge(...this.options.map((option) => option._optionHovered)),
+        ),
+        takeUntil(this._destroy),
+      )
+      .subscribe((a) => {
+        this._ngZone.run(() => {
+          this._keyManager.setActiveItem(a);
+        });
         this._resetOptions();
-        this._initializeSelection();
```

Comment:
Why do we remove the initializeSelection here. I guess this should still be the behaviour to set the first item active.